### PR TITLE
vdk-control-cli: clarify rest client errors

### PR DESCRIPTION
**Pull request by tozka**
We are making consequences configurable so that they are more precise on
per case basis. And also explain a bit more about countermeasures in
authorization error.

Testing Done: existing unit tests

Signed-off-by: Antoni Ivanov <aivanov@vmware.com>

### CODE REVIEW

File: projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py [hunk 1]
```diff
@@ -12,31 +12,38 @@
...
-                    countermeasures="Try to login again using VDK CLI login command. "
-                    "Make sure you have permission to execute the given operation.",
+                    what=self.__what,
+                    why=f"The request has not been applied because it lacks valid authentication credentials. "
+                    f"Error returned by control service is {exception.body}",
+                    consequences=self.__consequences,
+                    countermeasures="Try to login again using VDK CLI login command."
+                    "Or set correct api token configuration (see vdk config-help)."
+                    "Make sure you have permission to execute the given operation."
+                    "Investigate (google) the error returned by control service."
```
Comment (gageorgiev) [inline projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py:35]: nit: "(google)" seems a bit weird to include. Just "Investigate" is probably fine.

File: projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py [hunk 2]
```diff
@@ -12,31 +12,38 @@
...
+                    what=self.__what,
                     why="The requested resource cannot be found. "
                     "You may have a spelling mistake or the data job has not been created",
-                    consequences="Operation cannot complete.",
-                    countermeasures="Make sure that the data job name and team name  are spelled correctly (it is is case-sensitive) . "
-                    "Make sure you have run vdk create before doing any other operations on a data job.",
+                    consequences=self.__consequences,
+                    countermeasures="Make sure that the data job name and team name are spelled correctly "
+                    "(it is case-sensitive) . "
+                    "Make sure you have run vdk create --cloud before doing this operation on a data job.",
```
Comment (gageorgiev) [inline projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py:46]:
```suggestion
                    "Make sure you have run `vdk create --cloud` before doing this operation on a data job.",
```

Comment (gageorgiev) [reply projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py:46]: Apostrophes around the command makes it pop up when someone is skim reading the error.

**Review by gageorgiev — APPROVED** [review]

File: projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py (hunk 1 above)
Comment (tozka) [reply projects/vdk-control-cli/src/vdk/internal/control/rest_lib/rest_client_errors.py:35]: It's honest advice. A lot of people don't google enough :) Hopefully, someone finds it funny as well.

**Pull request merged**